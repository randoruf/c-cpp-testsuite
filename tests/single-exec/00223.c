
#include <stdio.h>


int recursive(int n){
    if (n <= 0) return 0;
    printf("Before call: %d\n", n); 
    int c = recursive(n - 1); 
    printf("After call: %d\n", c); 
    return c + 10;
}

int main(int argc, char** argv){
    printf("argc: %d\nargv: ", argc); 
    for (int i = 0; i < argc; i++){
        printf("%s, ", argv[i]);  
    }
    printf("\n-------------------------\n\n"); 
    recursive(254); 
}