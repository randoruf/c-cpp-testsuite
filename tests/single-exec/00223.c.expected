argc: 1
argv: ./a.out, 
-------------------------

Before call: 254
Before call: 253
Before call: 252
Before call: 251
Before call: 250
Before call: 249
Before call: 248
Before call: 247
Before call: 246
Before call: 245
Before call: 244
Before call: 243
Before call: 242
Before call: 241
Before call: 240
Before call: 239
Before call: 238
Before call: 237
Before call: 236
Before call: 235
Before call: 234
Before call: 233
Before call: 232
Before call: 231
Before call: 230
Before call: 229
Before call: 228
Before call: 227
Before call: 226
Before call: 225
Before call: 224
Before call: 223
Before call: 222
Before call: 221
Before call: 220
Before call: 219
Before call: 218
Before call: 217
Before call: 216
Before call: 215
Before call: 214
Before call: 213
Before call: 212
Before call: 211
Before call: 210
Before call: 209
Before call: 208
Before call: 207
Before call: 206
Before call: 205
Before call: 204
Before call: 203
Before call: 202
Before call: 201
Before call: 200
Before call: 199
Before call: 198
Before call: 197
Before call: 196
Before call: 195
Before call: 194
Before call: 193
Before call: 192
Before call: 191
Before call: 190
Before call: 189
Before call: 188
Before call: 187
Before call: 186
Before call: 185
Before call: 184
Before call: 183
Before call: 182
Before call: 181
Before call: 180
Before call: 179
Before call: 178
Before call: 177
Before call: 176
Before call: 175
Before call: 174
Before call: 173
Before call: 172
Before call: 171
Before call: 170
Before call: 169
Before call: 168
Before call: 167
Before call: 166
Before call: 165
Before call: 164
Before call: 163
Before call: 162
Before call: 161
Before call: 160
Before call: 159
Before call: 158
Before call: 157
Before call: 156
Before call: 155
Before call: 154
Before call: 153
Before call: 152
Before call: 151
Before call: 150
Before call: 149
Before call: 148
Before call: 147
Before call: 146
Before call: 145
Before call: 144
Before call: 143
Before call: 142
Before call: 141
Before call: 140
Before call: 139
Before call: 138
Before call: 137
Before call: 136
Before call: 135
Before call: 134
Before call: 133
Before call: 132
Before call: 131
Before call: 130
Before call: 129
Before call: 128
Before call: 127
Before call: 126
Before call: 125
Before call: 124
Before call: 123
Before call: 122
Before call: 121
Before call: 120
Before call: 119
Before call: 118
Before call: 117
Before call: 116
Before call: 115
Before call: 114
Before call: 113
Before call: 112
Before call: 111
Before call: 110
Before call: 109
Before call: 108
Before call: 107
Before call: 106
Before call: 105
Before call: 104
Before call: 103
Before call: 102
Before call: 101
Before call: 100
Before call: 99
Before call: 98
Before call: 97
Before call: 96
Before call: 95
Before call: 94
Before call: 93
Before call: 92
Before call: 91
Before call: 90
Before call: 89
Before call: 88
Before call: 87
Before call: 86
Before call: 85
Before call: 84
Before call: 83
Before call: 82
Before call: 81
Before call: 80
Before call: 79
Before call: 78
Before call: 77
Before call: 76
Before call: 75
Before call: 74
Before call: 73
Before call: 72
Before call: 71
Before call: 70
Before call: 69
Before call: 68
Before call: 67
Before call: 66
Before call: 65
Before call: 64
Before call: 63
Before call: 62
Before call: 61
Before call: 60
Before call: 59
Before call: 58
Before call: 57
Before call: 56
Before call: 55
Before call: 54
Before call: 53
Before call: 52
Before call: 51
Before call: 50
Before call: 49
Before call: 48
Before call: 47
Before call: 46
Before call: 45
Before call: 44
Before call: 43
Before call: 42
Before call: 41
Before call: 40
Before call: 39
Before call: 38
Before call: 37
Before call: 36
Before call: 35
Before call: 34
Before call: 33
Before call: 32
Before call: 31
Before call: 30
Before call: 29
Before call: 28
Before call: 27
Before call: 26
Before call: 25
Before call: 24
Before call: 23
Before call: 22
Before call: 21
Before call: 20
Before call: 19
Before call: 18
Before call: 17
Before call: 16
Before call: 15
Before call: 14
Before call: 13
Before call: 12
Before call: 11
Before call: 10
Before call: 9
Before call: 8
Before call: 7
Before call: 6
Before call: 5
Before call: 4
Before call: 3
Before call: 2
Before call: 1
After call: 0
After call: 10
After call: 20
After call: 30
After call: 40
After call: 50
After call: 60
After call: 70
After call: 80
After call: 90
After call: 100
After call: 110
After call: 120
After call: 130
After call: 140
After call: 150
After call: 160
After call: 170
After call: 180
After call: 190
After call: 200
After call: 210
After call: 220
After call: 230
After call: 240
After call: 250
After call: 260
After call: 270
After call: 280
After call: 290
After call: 300
After call: 310
After call: 320
After call: 330
After call: 340
After call: 350
After call: 360
After call: 370
After call: 380
After call: 390
After call: 400
After call: 410
After call: 420
After call: 430
After call: 440
After call: 450
After call: 460
After call: 470
After call: 480
After call: 490
After call: 500
After call: 510
After call: 520
After call: 530
After call: 540
After call: 550
After call: 560
After call: 570
After call: 580
After call: 590
After call: 600
After call: 610
After call: 620
After call: 630
After call: 640
After call: 650
After call: 660
After call: 670
After call: 680
After call: 690
After call: 700
After call: 710
After call: 720
After call: 730
After call: 740
After call: 750
After call: 760
After call: 770
After call: 780
After call: 790
After call: 800
After call: 810
After call: 820
After call: 830
After call: 840
After call: 850
After call: 860
After call: 870
After call: 880
After call: 890
After call: 900
After call: 910
After call: 920
After call: 930
After call: 940
After call: 950
After call: 960
After call: 970
After call: 980
After call: 990
After call: 1000
After call: 1010
After call: 1020
After call: 1030
After call: 1040
After call: 1050
After call: 1060
After call: 1070
After call: 1080
After call: 1090
After call: 1100
After call: 1110
After call: 1120
After call: 1130
After call: 1140
After call: 1150
After call: 1160
After call: 1170
After call: 1180
After call: 1190
After call: 1200
After call: 1210
After call: 1220
After call: 1230
After call: 1240
After call: 1250
After call: 1260
After call: 1270
After call: 1280
After call: 1290
After call: 1300
After call: 1310
After call: 1320
After call: 1330
After call: 1340
After call: 1350
After call: 1360
After call: 1370
After call: 1380
After call: 1390
After call: 1400
After call: 1410
After call: 1420
After call: 1430
After call: 1440
After call: 1450
After call: 1460
After call: 1470
After call: 1480
After call: 1490
After call: 1500
After call: 1510
After call: 1520
After call: 1530
After call: 1540
After call: 1550
After call: 1560
After call: 1570
After call: 1580
After call: 1590
After call: 1600
After call: 1610
After call: 1620
After call: 1630
After call: 1640
After call: 1650
After call: 1660
After call: 1670
After call: 1680
After call: 1690
After call: 1700
After call: 1710
After call: 1720
After call: 1730
After call: 1740
After call: 1750
After call: 1760
After call: 1770
After call: 1780
After call: 1790
After call: 1800
After call: 1810
After call: 1820
After call: 1830
After call: 1840
After call: 1850
After call: 1860
After call: 1870
After call: 1880
After call: 1890
After call: 1900
After call: 1910
After call: 1920
After call: 1930
After call: 1940
After call: 1950
After call: 1960
After call: 1970
After call: 1980
After call: 1990
After call: 2000
After call: 2010
After call: 2020
After call: 2030
After call: 2040
After call: 2050
After call: 2060
After call: 2070
After call: 2080
After call: 2090
After call: 2100
After call: 2110
After call: 2120
After call: 2130
After call: 2140
After call: 2150
After call: 2160
After call: 2170
After call: 2180
After call: 2190
After call: 2200
After call: 2210
After call: 2220
After call: 2230
After call: 2240
After call: 2250
After call: 2260
After call: 2270
After call: 2280
After call: 2290
After call: 2300
After call: 2310
After call: 2320
After call: 2330
After call: 2340
After call: 2350
After call: 2360
After call: 2370
After call: 2380
After call: 2390
After call: 2400
After call: 2410
After call: 2420
After call: 2430
After call: 2440
After call: 2450
After call: 2460
After call: 2470
After call: 2480
After call: 2490
After call: 2500
After call: 2510
After call: 2520
After call: 2530
